### ISSUE TITLE
Support for DOS 5.0 SHARE.EXE?

### ISSUE THREAD

**Issue by emendelson** [joncampbell123/dosbox-x#477]
Hello, and thank you for this terrific project. It seems as if you've incorporated the FAKESHAR feature that lets some programs think that SHARE.EXE is loaded, but this doesn't work with something like Word 6.0 for Windows 3.x, which won't install without the real SHARE.EXE. I know that DOSBox is for games, but if there's any hope of incorporating this, it would be nice to have. Thanks again...

**Comment by joncampbell123** [comment]
Quick idea: What is Word 6.0 looking for, that can be faked for a quick hack?

Long term: Which part of the SHARE API does it require and why?

**Comment by emendelson** [comment]
I wish I knew the answer to those questions! I'll see if I can find anything out and will let you know...

**Comment by sikthehedgehog** [comment]
Stupid question, but how does SHARE.EXE actually work? (i.e. how does it hook into DOS) Depending on the situation, it may be easier to just let it to work as intended. If it's just acting as a wrapper to the real DOS routines (i.e. it just calls back to DOS for the actual filesystem) then it may be able to just sit on DOSBox-X's own routines as long as it can hook.

As for Word: taking a wild guess that it's trying to open the file using whatever SHARE provides (instead of the normal DOS calls) and it throws the error when the call inevitably fails.

**Comment by joncampbell123** [comment]
As I understand it, SHARE.EXE is Microsoft's way of bolting a file locking API atop the DOS API so that multiple DOS programs can share a file and lock regions against changes from the other, especially in Windows.

**Comment by sikthehedgehog** [comment]
Found this:
http://www.techhelpmanual.com/691-int_2fh_1000h__is_share_exe_installed_.html
http://www.techhelpmanual.com/529-dos_fn_5ch__lock_unlock_file_access.html

Finding SHARE API documentation is harder than it should be. Also apparently Windows will hook over the interrupt call that's used for checking to always return true, which implies that merely running Windows should already be enough? Maybe there's some issue with replacing interrupt vectors?

> It sounds kind of dumb in a cooperative multitasking world like Windows 3.1 and DOS, but the API also supports locking across network shares as far as I know.

Also from DOS 4 onwards it ensures that old programs using FCB will still work on partitions larger than 32MB (to the point that DOS 4 will actively complain if it can't find SHARE.EXE by any means and the partition it's installed on is too large).

**Comment by emendelson** [comment]
I found (and now can't immediately find again, but I'll keep looking) a Microsoft document that says that Word uses Share.exe for OLE functions. It didn't go into any more detail, but presumably it locks (or tests for a lock) the file with which Word is communicating via OLE. But that's all I can guess at. I'll look again for that document.

Incidentally, there are people online who say that you can launch Word in DOSBox if vshare.386 is loaded, and/or if Share.exe is the Windows directory and autoexec.bat. I've tried all of these under Windows for Workgroups, and Word still tells me that Share.exe needs to be loaded.

**Comment by emendelson** [comment]
**_WAIT_** I was wrong - see the edited text of the first post! When I installed Windows for Workgroups 3.11, with VSHARE.386 and Word 6.0c, everything works correctly with DOSBox-X. With SVN, I get an error message saying that SHARE.EXE needs to be loaded - but there's no problem with DOSBox-X. 

I would have to experiment, but this seems to depend on WfW3.11 with VSHARE.386 - SHARE.EXE alone doesn't get the job done. But this means it's possible to run Word 6 - and presumably Office - under DOSBox-X.

**Comment by spz2020** [comment]
https://github.com/roncemer/freedos-share
or 
http://www.ibiblio.org/pub/micro/pc-stuff/freedos/files/distributions/1.2/repos/pkg-html/share.html

**Comment by Wengier** [comment]
SHARE is already supported for file-locking and record-locking etc as of the latest version.

**Comment by Coti4** [comment]
> SHARE is already supported for file-locking and record-locking etc as of the latest version.

Hi Wengier,

I installed DOSBox 0.74-3 version but I got SHARE not loaded" error. I copied SHARE file but still getting same error.

 What should I do to load  SHARE? Could you help me, please?

Thanks in advance,

**Comment by rderooy** [comment]
SHARE is supported in DOSBox-X which is a greatly enhanced fork of regular DOSBox. Regular DOSBox has no SHARE support.

Also in DOSBox-X, you do NOT load SHARE.EXE, the share support is built into DOSBox-X itself and can be controlled via a setting in the config file.

**Comment by Coti4** [comment]
Thanks rderooy!! I will try with DOSBox-X

Thanks for your quick response.
